keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/BakeryLifecycleMonitor.kt

```diff
@@ -71,6 +85,51 @@ class BakeryLifecycleMonitor(
       }
   }
 
+  /**
+   * Stores the ami details from a successful bake task so we don't have to wait for those
+   * details to appear in the clouddriver cache (~5 minute delay every time).
+   *
+   * If there's a problem parsing we log an error and move on, because we will see
+   * the image in clouddriver eventually.
+   */
+  suspend fun storeBakedImage(execution: ExecutionDetailResponse) {
+    // we know we launch a bake task with a stage called "bake".
+    val bakeStageRaw = execution.execution.stages?.find { it["type"] == "bake" }
+    if (bakeStageRaw == null) {
+      log.error("Trying to find baked ami information, but can't find a bake stage in execution ${execution.id}")
```

Comment:
Do we have the application name in the response at least to include in this log message?

Reply:
we do! i'll add it to all them. good call